bmi_sdk/bmi_sdk/bmi_variable.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import annotations
+
+from enum import Enum
+
+# Once python 3.8 is no longer supported, just get Annotated directly
+# from typing import Annotated, Any, Literal, Optional, Self, Union
+from typing import Any, Literal, Optional, Union
+
+import numpy as np
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+    Field,
+    RootModel,
+    computed_field,
+    model_validator,
+)
+
+# Only needed on python 3.8
+from typing_extensions import Annotated, Self
+
+from .bmi_grid import Grid
+
+Integer = RootModel[Annotated[str, Literal["int", "int16", "int32", "integer"]]]
```

Comment:
Let's drop these to a `RootModel[Literal[...]]` for simplicity.

Reply:
done in f6083a075fcd8e8594ff588bed910e2f3a524501